PSKoans/Koans/IntroTraining/AboutGetMember.koans.ps1

```diff
@@ -0,0 +1,137 @@
+
+    Make sure you use a different cmdlet each time, we'll be checking :)
+#>
+
+    It 'allows us to explore properties on an object' {
+
+        # Get some properties!
+        
+        <# EXAMPLE
+
+        $cmdlet1 = 'Get-Process'
+        $PropertyName = "Threads"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet1
+        Get-Process |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        #>
+        $BecauseString = "property '{0}' should be present in output from {1}"
+
+        $cmdlet1 = '__'
+        $PropertyName = "__"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet1
+        ____ |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $cmdlet2 = '__'
+        $PropertyName = "__"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet2
+        ____ |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $cmdlet3 = '__'
+        $PropertyName = "__"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet3
+        ____ |
+            Get-Member -MemberType Property -Name $PropertyName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $Cmdlets = $cmdlet1, $cmdlet2, $cmdlet3
+        $UniqueCheck = $Cmdlets | Get-Unique
+        $UniqueCheck.Count -eq 3 | Should -BeTrue -Because "three unique cmdlets should be supplied"
+
+    }
+
+    It 'allows us to explore methods on an object' {
+
+        # Get some methods!
+
+        <# EXAMPLE
+
+        $cmdlet1 = 'Get-Process'
+        $MethodName = "Close"
+        $Reason = $BecauseString -f $PropertyName, $cmdlet1
+        Get-Process |
+            Get-Member -MemberType Method -Name $MethodName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        #>
+
+        $BecauseString = "method '{0}' should be present in output from {1}"
+
+        $cmdlet1 = '__'
+        $MethodName = "__"
+        $Reason = $BecauseString -f $MethodName, $cmdlet1
+        ____ |
+            Get-Member -MemberType Method -Name $MethodName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $cmdlet2 = '__'
+        $MethodName = "__"
+        $Reason = $BecauseString -f $MethodName, $cmdlet2
+        ____ |
+            Get-Member -MemberType Method -Name $MethodName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $cmdlet3 = '__'
+        $MethodName = "__"
+        $Reason = $BecauseString -f $MethodName, $cmdlet3
+        ____ |
+            Get-Member -MemberType Method -Name $MethodName |
+            Should -Not -BeNullOrEmpty -Because $Reason
+
+        $Cmdlets = $cmdlet1, $cmdlet2, $cmdlet3
+        $UniqueCheck = $Cmdlets | Get-Unique
+        $UniqueCheck.Count -eq 3 | Should -BeTrue -Because "three unique cmdlets should be supplied"
+
+    }
+
```

Comment:
Blank line here and just above we should snip out. 🙂

There's one or two others a bit further up too.

Reply:
🦄